ocaml/typing/printtyp.ml

```diff
@@ -1570,13 +1570,21 @@ let param_jkind ty =
   | _ -> None (* this is (C2.2) from Note [When to print jkind annotations] *)
 
 let tree_of_label l =
-  let gom =
-    match l.ld_mutable, l.ld_global with
-    | Mutable, _ -> Ogom_mutable
-    | Immutable, Global -> Ogom_global
-    | Immutable, Unrestricted -> Ogom_immutable
+  let gbl =
+    match l.ld_global with
+    | Global -> Ogf_global
+    | Unrestricted -> Ogf_unrestricted
   in
-  (Ident.name l.ld_id, gom, tree_of_typexp Type l.ld_type)
+  let mut =
+    match l.ld_mutable with
+    | Immutable -> false
+    | Mutable m ->
+        if Misc.eq_from_le Alloc.Const.le m Alloc.Const.legacy then
+          true
+        else
+          Misc.fatal_errorf "Unexpected mutable(%a)" Alloc.Const.print m
```

Comment:
The printers in this file tend to print something like `"<foo>"` rather than erroring for cases that make semantic sense but have no syntax. That's slightly better when we use these printers for debugging the compiler.